crates/nix_health/src/check/direnv.rs

```diff
@@ -93,71 +133,99 @@ fn activation_check(local_flake: &std::path::Path, required: bool) -> Check {
     }
 }
 
-/// Information about a direnv install
-#[derive(Debug, Default, Serialize, Deserialize, Clone)]
-pub struct DirenvInstall {
-    /// Path to the direnv binary
-    pub bin_path: PathBuf,
-
-    /// Contents of `direnvrc`
-    pub direnv_config: Option<String>,
-
-    // bash_path used by direnv
-    pub bash_path: Option<PathBuf>,
-}
-
-impl DirenvInstall {
-    /// Detect user's direnv installation
+/// [Check] if direnv was already allowed in [project_dir]
 
-    pub fn detect() -> anyhow::Result<Self> {
-        let bin_path = which::which("direnv")?;
-        let output = std::process::Command::new(&bin_path)
-            .arg("status")
-            .output()?;
+fn is_direnv_allowed_on(project_dir: &std::path::Path) -> anyhow::Result<bool> {
```

Comment:
This can be moved as a method of `DirenvInstall` struct, no? This way we can run direnv using the exact detected path (`bin_path`) instead of searching $PATH again.